samcli/commands/build/command.py

```diff
@@ -124,6 +124,20 @@
     type=click.Path(),  # Must be a json file
     help="Path to environment variable json file (e.g., env_vars.json) to pass into build containers",
 )
+@click.option(
+    "--build-image",
+    "-bi",
+    default=None,
+    multiple=True,  # Can pass in multiple build images
+    required=False,
+    help="Container image URIs for building functions. "
+    "You can specify for all functions with just the image URI "
+    "(--build-image public.ecr.aws/sam/build-nodejs14.x:latest). "
```

Comment:
Why not have two options for this? We have seen in the past overriding options for different cases can become complex quickly. Seems like having one option for all and one option to specify per function might be safer to have. 

Side question: Did we consider putting this data on the function in the template Metadata like we do for other build information?

Reply:
This was done for consistency with the env var option as both of the have the same format and similar effect. We can bring this into UX review for a further discussion. 
Metadata in the template is not in scope of this change, but we can explore it in the future.

Reply:
I believe right now this is safe to have them in one option. Previously the container env var option have the the same overriding and precedence pattern which we feel that is safe and would not be too complex to cause any issues. Anyways we have a ux review set up on Friday as well, @jfuss if you can join to discuss this it would be nice as well!

Reply:
> Did we consider putting this data on the function in the template Metadata like we do for other build information?

I think this is a good call-out. We should have a consistent story about how parameters are passed. How do we choose between the template or CLI arguments? Why are CLI arguments are preferred here versus the template?